### ISSUE TITLE
Pipewire incompatibility?

### ISSUE THREAD

**Issue by birdie-github** [patroclos/PAmix#79]
Pressing volume up/down keys results in the Pipewire server completely freezing.

Audio continues playing but you cannot read anything off the server: current levels or anything.

Using:
Fedora Linux 40
pipewire-1.0.8-1.fc40.x86_64

I've tried the latest GIT snapshot, it's equally dysfunctional.

**Comment by patroclos** [comment]
Thanks for letting me know, I haven't really kept up with this project since I got a physical audio interface.
I'm currently doing a rewrite to C that's mostly complete and works on my pipewire setup, there's only a few non-essentials missing.
I'll try to get it ready for release by this weekend.

**Comment by patroclos** [comment]
I've just pushed the C rewrite to the dev branch, if you an please let me know if it works for you.
I changed the volume-bars to braille, so peak-monitors can now display much smaller changes, which is great for seeing noise or background-music.

**Comment by fucksophie** [comment]
C rewrite works flawlessly with Pipewire. Only issue is that arrow keys don't work on it :(

**Comment by patroclos** [comment]
@fucksophie arrow keys should work if you have them configured in `$HOME/.config/pamix.conf`.
Anyways, I added them to the default configuration (4de9ee1)

**Comment by fucksophie** [comment]
> @fucksophie arrow keys should work if you have them configured in `$HOME/.config/pamix.conf`. Anyways, I added them to the default configuration ([4de9ee1](https://github.com/patroclos/PAmix/commit/4de9ee17075408e1317879702f7cd1c36d562ee5))

I think dev and master can be merged, I'd love to have nixos packaging the dev version

**Comment by patroclos** [comment]
> I think dev and master can be merged, I'd love to have nixos packaging the dev version

Yeah, I'm pretty happy with the dev branch, just fixed the last issue I had with it.
I'll update the manpage and readme, then I can release it as v2.0
